### PR TITLE
Refactor system configuration

### PR DESCRIPTION
**Pull request by NSkelsey**
These changes address issue #999 and provide a solution for #1676

Notably the `Node` and `Notification` tables are no more. 

The migrations are in place and from this point onward it is a bug if they do not function.

@evilaliv3 I consider these changes ready to go. Obviously the timing on when to introduce them is a different decision.

### CODE REVIEW

File: backend/globaleaks/models/__init__.py [hunk 1]
```diff
@@ -104,16 +102,17 @@ def update(self, values=None):
...
-                if previous and isinstance(previous, dict):
-                    previous.update(value)
-                    setattr(self, k, previous)
-                else:
-                    setattr(self, k, value)
+                    if previous and isinstance(previous, dict):
+                        previous.update(value)
+                        setattr(self, k, previous)
+                    else:
+                        setattr(self, k, value)
```
Comment (NSkelsey) [inline backend/globaleaks/models/__init__.py]: This needs to go.

**Comment by vodkina** [comment]
Could you please add estimate to this issue? Thanks

File: backend/globaleaks/models/__init__.py (hunk 1 above)
Comment (NSkelsey) [inline backend/globaleaks/models/__init__.py]: removed in 79e3a5c

File: backend/globaleaks/db/sqlite.sql [hunk 2]
```diff
@@ -68,9 +87,12 @@ CREATE TABLE context (
     enable_two_way_comments INTEGER NOT NULL,
     enable_two_way_messages INTEGER NOT NULL,
     enable_attachments INTEGER NOT NULL,
-    status_page_message BLOB NOT NULL,
     presentation_order INTEGER,
     show_receivers_in_alphabetical_order INTEGER NOT NULL,
+    name TEXT NOT NULL,
+    description TEXT NOT NULL,
+    recipients_clarification TEXT NOT NULL,
+    status_page_message TEXT NOT NULL,
```
Comment (NSkelsey) [inline backend/globaleaks/db/sqlite.sql]: Bug, bug, bug

File: backend/globaleaks/db/sqlite.sql [hunk 3]
```diff
@@ -10,8 +32,8 @@ CREATE TABLE user (
     deletable INTEGER NOT NULL,
     role TEXT NOT NULL CHECK (role IN ('admin', 'receiver', 'custodian')),
     state TEXT NOT NULL CHECK (state IN ('disabled', 'enabled')),
+    description TEXT NOT NULL,
```
Comment (NSkelsey) [inline backend/globaleaks/db/sqlite.sql]: bug

File: backend/globaleaks/db/sqlite.sql [hunk 4]
```diff
@@ -531,3 +416,4 @@ CREATE INDEX field__template_id_index ON field(template_id);
 CREATE INDEX step__questionnaire_id_index ON step(questionnaire_id);
 CREATE INDEX context_questionnaire_id_index ON context(questionnaire_id);
 CREATE INDEX fieldanswer__internaltip_id_index ON fieldanswer(internaltip_id);
+CREATE INDEX config_group_index ON config(var_group);
```
Comment (NSkelsey) [inline backend/globaleaks/db/sqlite.sql]: Need to add another index for config_l10n

File: backend/globaleaks/handlers/admin/user.py [hunk 5]
```diff
@@ -87,8 +87,6 @@ def create_receiver_user(store, request, language):
 
 
 def db_create_user(store, request, language):
-    fill_localized_keys(request, models.User.localized_keys, language)
-
```
Comment (evilaliv3) [inline backend/globaleaks/handlers/admin/user.py]: @NSkelsey: i think this has been removed for error, do you confirm?

File: backend/globaleaks/handlers/public.py [hunk 6]
```diff
@@ -360,37 +327,35 @@ def _get_public_resources(store, language):
 
 
 class AhmiaDescriptionHandler(BaseHandler):
+    def _empt_gen(self):
```
Comment (evilaliv3) [inline backend/globaleaks/handlers/public.py]: @NSkelsey when not needed let's use explicit identifier or common abbreviations.

e.g. empt in place of empty is a not needed optimization (save only one char) that simply make the code less naturally readable.

File: backend/globaleaks/handlers/public.py [hunk 7]
```diff
@@ -360,37 +327,35 @@ def _get_public_resources(store, language):
...
-                                         serialize_node, self.request.language)
-
-        if node_info['ahmia']:
+        if GLSettings.memory_copy.ahmia:
             ret = yield GLApiCache.get('ahmia', self.request.language,
                                        serialize_ahmia, self.request.language)
 
             self.write(ret)
         else:  # in case of disabled option we return 404
+            yield self._empt_gen() # TODO remove inlineCallbacks from outer get.
```
Comment (evilaliv3) [inline backend/globaleaks/handlers/public.py]: @NSkelsey: this can be simply replaced with "yield None" or even "yield" :)

Magic!

**Comment by evilaliv3** [comment]
@NSkelsey: i applied really minor changes but i really enjoyed the rewrites you applied and the code quality :)

great pull request, well done!

**Pull request merged**